x-pack/plugins/observability_solution/infra/server/routes/entities/get_data_stream_types.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { type ObservabilityElasticsearchClient } from '@kbn/observability-utils/es/client/create_observability_es_client';
+import { type EntityClient } from '@kbn/entityManager-plugin/server/lib/entity_client';
+import { SOURCE_DATA_STREAM_TYPE } from '@kbn/observability-shared-plugin/common';
+import { CONTAINER_ID_FIELD, HOST_NAME_FIELD } from '../../../common/constants';
+import { type InfraMetricsClient } from '../../lib/helpers/get_infra_metrics_client';
+import { getHasMetricsData } from './get_has_metrics_data';
+import { getLatestEntity } from './get_latest_entity';
+
+interface Params {
+  entityId: string;
+  entityType: 'host' | 'container';
+  entityCentriExperienceEnabled: boolean;
+  infraMetricsClient: InfraMetricsClient;
+  obsEsClient: ObservabilityElasticsearchClient;
+  entityManagerClient: EntityClient;
+}
+
+export async function getDataStreamTypes({
+  entityCentriExperienceEnabled,
+  entityId,
+  entityManagerClient,
+  entityType,
+  infraMetricsClient,
+  obsEsClient,
+}: Params) {
+  const hasMetricsData = await getHasMetricsData({
+    infraMetricsClient,
+    entityId,
+    field: entityType === 'host' ? HOST_NAME_FIELD : CONTAINER_ID_FIELD,
+  });
+
+  const sourceDataStreams = new Set(hasMetricsData ? ['metrics'] : []);
```

Comment:
nit: The `EntityDataStreamType` from apm could be moved somewhere it could be used here